h2o-core/src/main/java/water/H2ONode.java

```diff
@@ -390,8 +408,19 @@ synchronized void freeTCPSocket( ByteChannel sock ) {
   // is specifically not any of the above channels.  This channel is limited to
   // messages which are presented in their entirety (not streamed) thus never
   // need another (nested) TCP channel.
-  private transient UDP_TCP_SendThread _sendThread = null; // set notnull if properly interned, and done before first sendMessage
-  public void sendMessage( ByteBuffer bb, byte msg_priority ) { _sendThread.sendMessage(bb,msg_priority); }
+  private transient UDP_TCP_SendThread _sendThread = null; // null if Node was removed from cloud
+  public final void sendMessage(ByteBuffer bb, byte msg_priority) {
+    UDP_TCP_SendThread sendThread = _sendThread;
+    if (sendThread == null) {
+      if (_removed_from_cloud) {
+        Log.warn("Node " + this + " is not active in the cloud anymore but we want to communicate with it." +
```

Comment:
H2O cannot get stuck this way because the client is the orchestrator of actions. We are never asking the client for anything, the client is always requesting something from us. In that sense, H2O cannot get stuck because the results of the tasks will always appear to be delivered to the H2O node (it is an asynchronous process).